catt/cli.py

```diff
@@ -68,85 +75,109 @@ def cast(device, video_url):
 
 
 @cli.command(short_help="Stop playing.")
-@pass_device
-def stop(device):
-    cast = CastController(device.name)
+@click.pass_obj
+def stop(settings):
+    cast = CastController(settings["device"])
     cast.kill()
 
 
 @cli.command(short_help="Rewind a video by SECS seconds.")
 @click.argument("seconds", type=click.INT, required=False, default=30, metavar="SECS")
-@pass_device
-def rewind(device, seconds):
-    cast = CastController(device.name)
+@click.pass_obj
+def rewind(settings, seconds):
+    cast = CastController(settings["device"])
     cast.rewind(seconds)
 
 
 @cli.command(short_help="Fastforward a video by SECS seconds.")
 @click.argument("seconds", type=click.INT, required=False, default=30, metavar="SECS")
-@pass_device
-def ffwd(device, seconds):
-    cast = CastController(device.name)
+@click.pass_obj
+def ffwd(settings, seconds):
+    cast = CastController(settings["device"])
     cast.ffwd(seconds)
 
 
 @cli.command(short_help="Seek the video to SECS seconds.")
 @click.argument("seconds", type=click.INT, metavar="SECS")
-@pass_device
-def seek(device, seconds):
-    cast = CastController(device.name)
+@click.pass_obj
+def seek(settings, seconds):
+    cast = CastController(settings["device"])
     cast.seek(seconds)
 
 
 @cli.command(short_help="Set the volume to LVL [0-1].")
 @click.argument("level", type=click.FLOAT, required=False, default=0.5, metavar="LVL")
-@pass_device
-def volume(device, level):
-    cast = CastController(device.name)
+@click.pass_obj
+def volume(settings, level):
+    cast = CastController(settings["device"])
     cast.volume(level)
 
 
 @cli.command(short_help="Turn up volume by an 0.1 increment.")
-@pass_device
-def volumeup(device):
-    cast = CastController(device.name)
+@click.pass_obj
+def volumeup(settings):
+    cast = CastController(settings["device"])
     cast.volumeup()
 
 
 @cli.command(short_help="Turn down volume by an 0.1 increment.")
-@pass_device
-def volumedown(device):
-    cast = CastController(device.name)
+@click.pass_obj
+def volumedown(settings):
+    cast = CastController(settings["device"])
     cast.volumedown()
 
 
 @cli.command(short_help="Show some information about the currently-playing video.")
-@pass_device
-def status(device):
-    cast = CastController(device.name)
+@click.pass_obj
+def status(settings):
+    cast = CastController(settings["device"])
     cast.status()
 
 
 @cli.command(short_help="Show complete information about the currently-playing video.")
-@pass_device
-def info(device):
-    cast = CastController(device.name)
+@click.pass_obj
+def info(settings):
+    cast = CastController(settings["device"])
     cast.info()
 
+@click.pass_obj
+def writeconfig(settings):
+    config_dir = click.get_app_dir("catt")
+    try:
+        os.mkdir(config_dir)
+    except FileExistsError:
+        pass
+    config_filename = os.path.join(config_dir, "catt.json")
+    with open(config_filename, "w") as config:
+        json.dump(settings, config)
```

Comment:
I'm not loving JSON for configuration, I generally prefer ini-style files with the built-in `ConfigParser`.